File: silver/25206/python/main.py

if (__name__ == "__main__"):
    sumScore = 0.0; sumCredit = 0.0
    creditlist = [4.5, 4.0, 3.5, 3.0, 2.5, 2.0, 1.5, 1.0]

    for i in range(20):
        text, credit, score = input().split(" ")
        if (score == 'P'): continue
        if (score == 'F'): sumCredit += float(credit); continue

        index = (ord(score[0]) - 65) * 2
        index += 0 if (score[1] == '+') else 1
        sumScore += creditlist[index] * float(credit)
        sumCredit += float(credit)

    print(sumScore / sumCredit)

File: silver/25206/c++/main.cpp
#include <iostream>
#include <string>
#include <cstdio>

int main()
{

    float creditVector[8] = {4.5, 4.0, 3.5, 3.0, 2.5, 2.0, 1.5, 1.0}; // F학점 제외
    float totalScore = 0.0, credit;
    
    float sumScore = 0.0, sumCredit = 0.0;

    int index;
    std::string score;

    for (int i = 0 ; i < 20; i++) {
        std::string text;
        std::cin >> text >> credit >> score;
        if (score == "P") { continue; }
        if (score == "F") { sumCredit += credit; continue; }
        index = ((static_cast<int>(score[0]) - 65) * 2);
        index += score[1] == '+' ? 0 : 1;
        sumScore += creditVector[index] * credit;
        sumCredit += credit;
    }

    printf("%.6f", (sumScore / sumCredit));

    return 0;
}